utils/ort/src/test/kotlin/VersionUtilsTest.kt

```diff
@@ -247,5 +246,23 @@ class VersionUtilsTest : WordSpec({
 
             filterVersionNames("6.2.9.Final", names) should containExactly("6.2.9")
         }
+
+        "treat full matches higher than near exact matches" {
+            val names = listOf(
+                "1.0.0",
+                "v1.0.0"
+            )
+
+            filterVersionNames("1.0.0", names) shouldHaveSingleElement "1.0.0"
```

Comment:
Hmm. Doesn't this go against what you wrote in the commit message:

> Therefore, a version using this prefix should be treated with a higher
certainty than others.

I was reading this as an exact match with a "v" prefix should now have highest priority, as "v" indicates it's a version (as opposed to some other number).